File: DevExpress-Demo/DevExpress.ProductsDemo.Wpf/VB/DevExpress.ProductsDemo.Wpf/Controls/InfoViewModel.vb
Imports System.Windows.Input
Imports DevExpress.Xpf
Imports DevExpress.Mvvm
Imports DevExpress.Utils

Namespace ProductsDemo

    Public Class InfoViewModel
        Inherits BindableBase

        Private aboutInfoField As AboutInfo

        Private showHelpCommandField As ICommand

        Private showGettingStartedCommandField As ICommand

        Private showBuyNowCommandField As ICommand

        Public ReadOnly Property AboutInfo As AboutInfo
            Get
                If aboutInfoField Is Nothing Then
                    aboutInfoField = New AboutInfo()
                    Dim data As DevExpress.Internal.UserData = DevExpress.Utils.About.Utility.GetInfoEx()
                    aboutInfoField.LicenseState = If(data.IsExpired, LicenseState.TrialExpired, If(data.IsTrial, LicenseState.Trial, LicenseState.Licensed))
                    aboutInfoField.ProductName = "WPF Controls"
                    aboutInfoField.ProductPlatform = "Build Your Own Office"
                    aboutInfoField.RegistrationCode = WpfSerialNumberProvider.GetSerial()
                    aboutInfoField.Version = AssemblyInfo.Version
                End If

                Return aboutInfoField
            End Get
        End Property

        Public Sub ShowHelp()
            Core.DocumentPresenter.OpenLink(AssemblyInfo.DXLinkGetSupport)
        End Sub

        Public Sub ShowGettingStarted()
            Core.DocumentPresenter.OpenLink("https://go.devexpress.com/Demo_GetStarted_WPF.aspx")
        End Sub

        Public Sub ShowBuyNow()
            Core.DocumentPresenter.OpenLink("https://go.devexpress.com/Demo_Subscriptions_Buy.aspx")
        End Sub

        Public ReadOnly Property ShowHelpCommand As ICommand
            Get
                If showHelpCommandField Is Nothing Then showHelpCommandField = New DelegateCommand(AddressOf ShowHelp, False)
                Return showHelpCommandField
            End Get
        End Property

        Public ReadOnly Property ShowGettingStartedCommand As ICommand
            Get
                If showGettingStartedCommandField Is Nothing Then showGettingStartedCommandField = New DelegateCommand(AddressOf ShowGettingStarted, False)
                Return showGettingStartedCommandField
            End Get
        End Property

        Public ReadOnly Property ShowBuyNowCommand As ICommand
            Get
                If showBuyNowCommandField Is Nothing Then showBuyNowCommandField = New DelegateCommand(AddressOf ShowBuyNow, False)
                Return showBuyNowCommandField
            End Get
        End Property
    End Class
End Namespace

File: DevExpress-Demo/DevExpress.ProductsDemo.Wpf/CS/DevExpress.ProductsDemo.Wpf/Controls/InfoViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Input;
using DevExpress.Xpf;
using DevExpress.Mvvm;
using DevExpress.Utils;

namespace ProductsDemo {
    public class InfoViewModel : BindableBase {
        AboutInfo aboutInfo;
        ICommand showHelpCommand;
        ICommand showGettingStartedCommand;
        ICommand showBuyNowCommand;

        public AboutInfo AboutInfo {
            get {
                if(aboutInfo == null) {
                    aboutInfo = new AboutInfo();
                    DevExpress.Internal.UserData data = DevExpress.Utils.About.Utility.GetInfoEx();                    
                    aboutInfo.LicenseState = data.IsExpired ? LicenseState.TrialExpired : data.IsTrial ? LicenseState.Trial : LicenseState.Licensed;
                    aboutInfo.ProductName = "WPF Controls";
                    aboutInfo.ProductPlatform = "Build Your Own Office";
                    aboutInfo.RegistrationCode = WpfSerialNumberProvider.GetSerial();
                    aboutInfo.Version = AssemblyInfo.Version;
                }
                return aboutInfo;
            }
        }
        public void ShowHelp() {
            DevExpress.Xpf.Core.DocumentPresenter.OpenLink(AssemblyInfo.DXLinkGetSupport);
        }
        public void ShowGettingStarted() {
            DevExpress.Xpf.Core.DocumentPresenter.OpenLink("https://go.devexpress.com/Demo_GetStarted_WPF.aspx");
        }
        public void ShowBuyNow() {
            DevExpress.Xpf.Core.DocumentPresenter.OpenLink("https://go.devexpress.com/Demo_Subscriptions_Buy.aspx");
        }
        public ICommand ShowHelpCommand {
            get {
                if(showHelpCommand == null)
                    showHelpCommand = new DelegateCommand(ShowHelp, false);
                return showHelpCommand;
            }
        }
        public ICommand ShowGettingStartedCommand {
            get {
                if(showGettingStartedCommand == null)
                    showGettingStartedCommand = new DelegateCommand(ShowGettingStarted, false);
                return showGettingStartedCommand;
            }
        }
        public ICommand ShowBuyNowCommand {
            get {
                if(showBuyNowCommand == null)
                    showBuyNowCommand = new DelegateCommand(ShowBuyNow, false);
                return showBuyNowCommand;
            }
        }
    }
}
